symengine/symengine_rcp.h

```diff
@@ -198,6 +198,73 @@ using Teuchos::print_stack_on_segfault;
 #endif
 
 
+template <class T>
+class EnableRCPFromThis {
+// Public interface
+public:
+    //! Get RCP<T> pointer to self (it will cast the pointer to T)
+    inline RCP<T> rcp_from_this() {
+#if defined(WITH_SYMENGINE_RCP)
+        return rcp(static_cast<T*>(this));
+#else
+        return rcp_static_cast<T>(weak_self_ptr_.create_strong());
+#endif
+    }
+
+    //! Get RCP<const T> pointer to self (it will cast the pointer to const T)
+    inline RCP<const T> rcp_from_this() const {
```

Comment:
@bartlettroscoe thanks for the feedback. I would like to make it const correct, but I am a bit struggling what the right way to do this is. What I don't understand is that the C++ standard itself is using the following two functions (see http://en.cppreference.com/w/cpp/memory/enable_shared_from_this/shared_from_this):
```
shared_ptr<T> shared_from_this(); 	            (1) 	
shared_ptr<T const> shared_from_this() const;	(2) 
```
and I am just doing what they do. So it seems if there is a problem with this, it's already in the C++ standard.